src/logic/reminders.ts

```diff
@@ -0,0 +1,82 @@
+import { isTest } from '../utils/env';
+
+const REMINDER_ENDPOINT_PROD = 'https://support.theguardian.com/reminders/create/one-off';
+// const REMINDER_ENDPOINT_CODE = 'https://support.code.dev-theguardian.com/reminders/create/one-off';
+
+export type ReminderStatus = 'DEFAULT' | 'IN_PROGRESS' | 'SUCCESS' | 'FAILURE';
+
+export type ReminderPlatform = 'WEB' | 'AMP';
+
+export type ReminderComponent = 'EPIC' | 'BANNER';
+
+// The oneoff reminder will only be set for unrecognised emails if reminderStage === PRE
+// Not an issue in Prod because only signed in users will see the banner or epic
+// - but can cause heartbreak while testing locally with an unrecognised email address
+export type ReminderStage = 'PRE' | 'POST' | 'WINBACK';
+
+export interface BaseSignupRequest {
+    email: string;
+    reminderPlatform: ReminderPlatform;
+    reminderComponent: ReminderComponent;
+    reminderStage: ReminderStage;
+    country?: string;
+    reminderOption?: string;
+}
+
+export interface OneOffSignupRequest extends BaseSignupRequest {
+    reminderPeriod: string;
+}
+
+const getReminderDate = (date: Date): Date => {
+    const monthsAhead = date.getDate() < 20 ? 1 : 2;
+    date.setMonth(date.getMonth() + monthsAhead);
+
+    return date;
+};
+
+export interface ReminderFields {
+    reminderCta: string;
+    reminderLabel: string;
+    reminderPeriod: string;
+    reminderOption?: string;
+}
+
+export const buildReminderFields = (today: Date = new Date()): ReminderFields => {
```

Comment:
Given that the logic here is slightly tricky, is it worth adding some jest tests to act as documentation (and give us confidence if we ever want to make changes to this code)?

Reply:
I can have a go ...

Reply:
Done - let me know if we need to test other things here